gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -266,6 +288,35 @@ private InsertObjectRequest buildInsertRequest(
       return requestBuilder.build();
     }
 
+    // Handles the case when a writeOffset of data read previously is being processed.
+    // This happens if a transient failure happens while uploading, and can be resumed by
+    // querying the current committed offset.
+    private InsertObjectRequest buildRequestFromCachedDataChunk(
```

Comment:
How about calling this buildRequestFromBufferedDataChunk? Caching makes people wonder about cache consistency, whereas here you're really just operating off of disk buffers.

Reply:
Make sense, updated.